packages/block-library/src/paragraph/edit.js

```diff
@@ -108,6 +123,36 @@ function ParagraphBlock( {
 					</ToolsPanelItem>
 				</InspectorControls>
 			) }
+			{ ! content && (
+				<Popover
+					anchorRef={ paragraphElement }
+					animate={ false }
+					position="top right left"
+					focusOnMount={ false }
+					__unstableForcePosition
+					className="wp-block-paragraph__drop_zone"
+				>
+					<DropZone
```

Comment:
I do think it'd be interesting to try `useDropZone`, and only render the visual part of the dropzone when the user is actually dragging a file over.

That way you don't need the `pointer-events` change, and possibly also won't need to listen for resize of the paragraph.